### DISCUSSION TITLE
Steering Committee Elections

### DISCUSSION THREAD

**Discussion by leggetter** [DevRel-Foundation/governance#93] [Ideas]
> [!NOTE]
> **This initial post is a work-in-progress and will be updated based on feedback and to reflect the current state of the discussion**

The current Developer Relations Foundation (DRF) Steering Committee (SC) consists of a group of 5 volunteers from the following groups:

a) People who started the project to form the DRF
b) People who were invited by those in a) using a skills matrix assessment

In the future, the SC will consist of people voted in by the DRF community, with the potential for some positions to be appointed, to ensure the SC is diverse and correctly represents the developer relations community globally.

The purpose of this discussion is to define:

1. The state of the current SC
2. What future state of the SC should look like
3. How people become members of the SC - the SC election process

### References

- [Kubernetes Steering Committee Election, Voters Guide 2024](https://github.com/kubernetes/community/tree/master/elections/steering/2024)

**Comment by anajsana** [comment]
Thanks for opening this @leggetter ! Maybe we can allocate time during today's APAC & EMEA community call to discuss further and see what folks's ideas, and feedback are 👍 Just a few thoughts from my side:

1. Whatever decision the DRF community makes, I believe it’s important that the details are included in the first version of the [DRF charter](https://github.com/DevRel-Foundation/governance/blob/main/Technical_Charter_v1.0.adoc), as there is currently no documentation on the composition.

2. Regarding the DRF Steering Committee (SC) composition and voting process:

I think there are some useful sections in the Kubernetes SC document that we could adapt. However, while I understand that a large project like Kubernetes has many considerations, I personally wouldn’t replicate all of it. I feel that some parts might be unnecessary for DRF at this early stage, and I’d suggest opting for a simpler process. 

With that in mind, I’d like to propose considering the [composition that the TODO Group charter](https://github.com/DevRel-Foundation/governance/blob/main/Technical_Charter_v1.0.adoc) follows as a potential model. Perhaps a blend of both processes could be a good starting point.

> 
> 
> _The Steering Committee shall be composed of seven (7) General Members, with five (5) elected and two (2) appointed._
> 
> - _Committee member terms shall begin on January 1st of the year following election or appointment, and terminate on December 31st of the second year following election or appointment._
> 
> - _In each calendar year, the group shall hold an election to fill elected positions which will be vacated at the end of that year. The election process shall include nominations and voting during the month of September._
> 
> - _In each calendar year, the committee shall appoint one (1) member to fill the appointed position which will be vacated at the end of that year. The appointment shall be made during the month of October, by the current members of the commitee at that time._
> 
> - _In even-numbered years, two (2) elected positions and one (1) appointed position shall be filled by the election and appointment processes._
> 
> - _In odd-numbered years, three (3) elected positions and one (1) appointed position shall be filled by the election and appointment processes._
> 
> - _If any committee member should resign or otherwise be unable to complete their term for any reason, with more than six (6) months remaining in their term, the remaining committee members may appoint a replacement member. For election or appointment purposes, that member’s term will end when the outgoing member’s term would have ended._
> 
> - _If any committee member should resign or otherwise be unable to complete their term for any reason, with six (6) months or fewer remaining in their term, their position will be left unfilled until the regularly scheduled election or appointment process for that position._
> 
> - _TODO Steering Committee members are elected for a two year term._
> 
> - _Elections shall be run using the Condorcet-IRV method through the Cornell online service (http://civs.cs.cornell.edu/). The methodology or service used in future elections may be changed via a five (5) member approval vote of current members of the committee at that time._

Re: Candidacy. I also believe that the process from the Kubernetes project is designed for a fairly large contributor community. As another example, in [TODO](https://github.com/todogroup/governance/tree/main?tab=readme-ov-file#mission-and-decision-making), the candidacy process is documented as follows:

> 
> _The Steering Committee Election happens every year during October and anyone can become a candidate by submitting the [application form](https://forms.gle/CXgUJhjuXYqsjsXr6) before October 10th, 2024._
> 
> _While participation is open to all, individuals who actively engage with the TODO community or projects are more likely to be elected._
> 

If there are other projects that the DRF community is aware of that could serve as good examples for DRF context, I’d love to hear about them! Please feel free to share any links in the comments.

**Reply by divya-mohan0209** [reply]
There's also the k/steering election charter which is slightly more technical, but we could borrow from: https://github.com/kubernetes/steering/blob/main/elections.md

One of the salient points that I'd like us to borrow is that of the election officer/s. While we're formalising the group of community managers, since governance is *a part* of community management, I'd recommend we have the managers/maintainers of the group take on the onus of election officers. This would help us ensure the due diligence criteria we're listing out in our proposal are met by our candidates and these are also verified by officers outside of the existing steering committee to avoid bias.

**Reply by anajsana** [reply]
Just wanted to highlight @divya-mohan0209’s idea here! I think it’s a great template for DRF to use. I also found this high-level template that could work as a general SC governance document aside from this one, also based on CNCF templates.

https://github.com/cncf/project-template/blob/main/GOVERNANCE-elections.md

Would love to hear your thoughts!

**Comment by leggetter** [comment]
## Draft Proposal (based on TODO Group)

The Developer Relations Foundation (DRF) Steering Committee Election happens every year during October and anyone can become a candidate by submitting an **application form** before October 10th.

While participation is open to all, individuals who actively engage with the DRF community or projects are more likely to be elected. Active participation or contribution in DRF includes: (1) Supporting DRF as a Manager or Participant OR (2) Acting as a Maintainer and/or contributing actively to any of the DRF GitHub repositories. OR (3) Hosting local events supported by DRF.

There are two types of seats: Elected seats and Appointed seats. Appointed seats exist as a mechanism to ensure diversity in terms of region, gender, and industry.

- Elected seats: Each DRF General Member's main contact reviews candidate applications and casts their votes during October. Elected Candidates will be notified in November, and the SC onboarding period will begin.
- Appointed seats: Current Steering Committee members appoint a candidate from the 2025 list.

## DRF Steering Committee Composition

The Steering Committee shall be composed of seven (7) Participants, with five (5) elected and two (2) appointed.

Committee member terms shall begin on January 1st of the year following election or appointment, and terminate on December 31st of the second year following election or appointment.

In each calendar year, the group shall hold an election to fill elected positions which will be vacated at the end of that year. The election process shall include nominations and voting during the month of September.

In each calendar year, the committee shall appoint one (1) member to fill the appointed position which will be vacated at the end of that year. The appointment shall be made during the month of October, by the current members of the commitee at that time.

In even-numbered years, two (2) elected positions and one (1) appointed position shall be filled by the election and appointment processes.

In odd-numbered years, three (3) elected positions and one (1) appointed position shall be filled by the election and appointment processes.

If any committee member should resign or otherwise be unable to complete their term for any reason, with more than six (6) months remaining in their term, the remaining committee members may appoint a replacement member. For election or appointment purposes, that member’s term will end when the outgoing member’s term would have ended.

If any committee member should resign or otherwise be unable to complete their term for any reason, with six (6) months or fewer remaining in their term, their position will be left unfilled until the regularly scheduled election or appointment process for that position.

DRF Steering Committee members are elected for a two year term.

Elections shall be run using the Condorcet-IRV method through the Cornell online service (http://civs.cs.cornell.edu/). The methodology or service used in future elections may be changed via a five (5) member approval vote of current members of the committee at that time.

---

## Notes/Questions

- Questions for the first all-appointed SC group
  - When did/does the current SC members tenure begin?
  - Given the 2 and 3 elected / appointed rotation process per year, current SC members may need to step down early
- Types of contributors:
  - DRF current has mention of a **Manager** and a **Participant**. Are there other types, such as a **Maintainer**? Maybe someone who isn't involved in a working group but has contributed to a body of work in GitHub?
  - What about participation within Discord?
  - Do we need these more clearly defined?

**Comment by katiewaz1977** [comment]
Overall thumb's up, with a request for more clarity on the appointed positions. I'd like to make sure we have consensus on the dimensions of diversity we're considering, as well as clarity on what the "2025 list" is. Thank you for this work!

**Reply by leggetter** [reply]
Ah, the 2025 list is a copy and paste and I'm unsure what that is in the TODO Group. Maybe @anajsana  can clarify?

**Reply by anajsana** [reply]
I guess you are referring to this "Appointed seats: Current Steering Committee members appoint a candidate from the 2025 list" correct?

In TODO, we run a Call for Nominations form 2–3 weeks before the election poll opens. During this time, anyone interested can apply to become a candidate for the TODO Elections. The form includes the following fields:

- Participant Name, years of experience in open source management or similar roles, affiliation, bio, and profile picture
- Pronouns and geographical location
- A statement on why they wish to serve TODO as a Steering Committee member, how they plan to contribute, and what they have already contributed to the TODO community (e.g, presenting at conferences, contributing to the repository, participating in community calls or discussion forums, organizing OSPO local meetings in their regions, or serving as a TODO regional ambassador)

Thanks to this info, we are then able to create:

- A blog post on the TODO website that introduces the [insert year] candidates (see this year's example: https://todogroup.org/blog/sc-elections-2025/)
- A PDF that is attached in the email sent when the election poll starts to those people eligible to vote
- A social to promote in SM channels and TODO Community ML and slack
Examples below: 

![2025candidates social](https://github.com/user-attachments/assets/6760bb94-63f8-4650-9dab-6b5c43174d6f)

[2025 TODO SC Candidates List](https://github.com/user-attachments/files/17792430/2025SC-Candidates-1-1.pdf)


I think this process is easy to copy-paste for DRF use case too. Just let me know your thoughts and feel welcome to chime in if there are other ideas! Everyone is welcome to take part in this conversation!

**Comment by anajsana** [comment]
Found this CNCF Steering Committee Elections Template quite interesting to use: https://contribute.cncf.io/maintainers/templates/governance-elections/
